### PR TITLE
added unsorted_segment_max to ivy functional API

### LINKED ISSUE #19584: unsorted_segment_max

### PR DESCRIPTION
**Pull request by Nishu0**
Close #19584

### CODE REVIEW

**Comment by Nishu0** [comment]
Hey 👋 @MahmoudAshraf97 @hmahmood24 can you review and let me know if any changes are required

**Comment by Nishu0** [comment]
Hey 👋 @MahmoudAshraf97 @hmahmood24  can you review and let me know if any changes are required

File: ivy/data_classes/container/experimental/creation.py
Comment (hmahmood24) [inline ivy/data_classes/container/experimental/creation.py:868]: This indentation error is causing the CI tests to fail. Can you kindly fix these?

**Review by hmahmood24 — CHANGES_REQUESTED** [review]
Hey @Nishu0, thank you for this PR. Apologies for being late in responding. There are some lint/formatting issues with this PR: https://github.com/unifyai/ivy/actions/runs/5591899948/job/15146081348?pr=19585. Can you kindly fix this along with the suggested change? Thanks!

File: ivy/data_classes/container/experimental/creation.py
Comment (Nishu0) [reply ivy/data_classes/container/experimental/creation.py]: Hey @hmahmood24 Thanks for reviewing. I have solved the error. Let me know if there are any other changes required.

**Review by hmahmood24 — CHANGES_REQUESTED** [review]
@Nishu0 Can you please fix the merge conflicts first so that the CI tests can run on your PR? Thanks!

**Comment by hmahmood24** [comment]
Hey @Nishu0. Apologies for the late response. I just resolved some merge conflicts and now waiting for the CI tests to run then I'll leave a review accordingly. Thanks!

**Review by hmahmood24 — CHANGES_REQUESTED** [review]
@Nishu0 It seems like the test you added is [failing](https://github.com/unifyai/ivy/actions/runs/6065633210/job/16455346763?pr=19585) for all backends. The PR has some [lint/formatting issues](https://github.com/unifyai/ivy/actions/runs/6065633218/job/16455347347?pr=19585) you'll need to look into. Thanks! :)

**Comment by Nishu0** [comment]
> @Nishu0 It seems like the test you added is [failing](https://github.com/unifyai/ivy/actions/runs/6065633210/job/16455346763?pr=19585) for all backends. The PR has some [lint/formatting issues](https://github.com/unifyai/ivy/actions/runs/6065633218/job/16455347347?pr=19585) you'll need to look into. Thanks! :)

Hey @hmahmood24 Thanks for the Feedback, I tried restoring the old code of the test from the main branch of Ivy as you can see it's still failing there is a Linting error from the before code. Let me know if any others changes are required from my end.

**Review by hmahmood24 — CHANGES_REQUESTED** [review]
@Nishu0 I think you also reverted the commits that added the unit test? Also for the lint checks, you can use `pre-commit` to perform auto-linting or give you helpful feedback on changes needed to make to get the lint checks on the CI passing

**Comment by hmahmood24** [comment]
Hey @Nishu0, not sure if you're actively working on this PR or not but best to request for a review once you resolve the merge conflicts and are confident that the tests are passing so that this PR doesn't go unnoticed and we can get it merged ASAP. Thanks :)

**Comment by hmahmood24** [comment]
Closing this PR since it has been stale from more than 2 weeks now.

**Pull request closed without merging**
